keras_nlp/models/distilbert/distilbert_preprocessing.py

```diff
@@ -18,78 +18,80 @@
-
-# Map a dataset to preprocess a single sentence.
-features = ["The quick brown fox jumped.", "I forgot my homework."]
-labels = [0, 1]
-ds = tf.data.Dataset.from_tensor_slices((features, labels))
-ds = ds.map(
-    lambda x, y: (preprocessor(x), y),
-    num_parallel_calls=tf.data.AUTOTUNE,
-)
-
-# Map a dataset to preprocess a multiple sentences.
-first_sentences = ["The quick brown fox jumped.", "Call me Ishmael."]
-second_sentences = ["The fox tripped.", "Oh look, a whale."]
-labels = [1, 1]
-ds = tf.data.Dataset.from_tensor_slices(((first_sentences, second_sentences), labels))
-ds = ds.map(
-    lambda x, y: (preprocessor(x), y),
-    num_parallel_calls=tf.data.AUTOTUNE,
-)
-```
-"""
-
 
 @keras.utils.register_keras_serializable(package="keras_nlp")
 class DistilBertPreprocessor(keras.layers.Layer):
+    """DistilBERT preprocessing layer.
+
+    This preprocessing layer will do three things:
+
+     - Tokenize any number of inputs using a
+       `keras_nlp.tokenizers.WordPieceTokenizer`.
+     - Pack the inputs together using a `keras_nlp.layers.MultiSegmentPacker`.
+       with the appropriate `"[CLS]"`, `"[SEP]"` and `"[PAD]"` tokens.
+     - Construct a dictionary of with keys `"token_ids"`, `"padding_mask"`,
+       that can be passed directly to a DistilBERT model.
+
+    This layer will accept either a tuple of (possibly batched) inputs, or a
+    single input tensor. If a single tensor is passed, it will be packed
+    equivalently to a tuple with a single element.
+
+    The WordPiece tokenizer can be accessed via the `tokenizer` property on this
+    layer, and can be used directly for custom packing on inputs.
+
+    Args:
+        vocabulary: A list of vocabulary terms or a vocabulary filename.
+        lowercase: If `True`, input will be lowercase before tokenization. If
+            `vocabulary` is set to a pretrained vocabulary, this parameter will
+            be inferred.
+        sequence_length: The length of the packed inputs. Only used if
+            `pack_inputs` is True.
+        truncate: The algorithm to truncate a list of batched segments to fit
+            within `sequence_length`. Only used if
+            `pack_inputs` is True. The value can be either `round_robin` or
+            `waterfall`:
+                - `"round_robin"`: Available space is assigned one token at a
+                    time in a round-robin fashion to the inputs that still need
+                    some, until the limit is reached.
+                - `"waterfall"`: The allocation of the budget is done using a
+                    "waterfall" algorithm that allocates quota in a
+                    left-to-right manner and fills up the buckets until we run
+                    out of budget. It supports an arbitrary number of segments.
+
+    Examples:
+    ```python
+    preprocessor = keras_nlp.models.DistilBertPreprocessor(
+        vocabulary="./vocab.txt",
+    )
+
+    # Tokenize and pack a single sentence directly.
```

Comment:
Does this line actually work? We had an open question about this in #395

Reply:
Yes, it does I believe, it creates an unbatched input, which is the correct behavior in that case. The question we need to resolve on that issue is how a model should respond to being passed an unbatched input. Can respond on the issue.

For now, this change would leave us consistent with BERT, so that seems ideal as we continue discussion.